web_monitoring/db.py

```diff
@@ -1,47 +1,608 @@
+
+
+@ensure_credentials
+def get_page(page_id):
+    """
+    Lookup a specific Page by ID.
+
+    Parameters
+    ----------
+    page_id : string
+
+    Returns
+    -------
+    response : dict
+    """
+    url = f'{api_url()}/pages/{page_id}'
+    res = requests.get(url, auth=auth())
+    res.raise_for_status()
+    result = res.json()
+    # In place, replace datetime strings with datetime objects.
+    data = result['data']
+    data['created_at'] = parse_timestamp(data['created_at'])
+    data['updated_at'] = parse_timestamp(data['updated_at'])
+    if 'latest' in data:
+        data['latest']['capture_time'] = parse_timestamp(
+            data['latest']['capture_time'])
+    return result
+
+
+### VERSIONS ###
+
+
+@ensure_credentials
+def list_page_versions(page_id, chunk=None, chunk_size=None,
+                       start_date=None, end_date=None,
+                       source_metadata=None):
+    """
+    List Versions for a given Page, optionally filtered by search criteria.
+
+    Parameters
+    ----------
+    page_id : string
+    chunk : integer, optional
+        pagination parameter
+    chunk_size : integer, optional
+        number of items per chunk
+    start_date : datetime, optional
+    end_date : datetime, optional
+    source_metadata : dict, optional
+        Examples:
+
+        * ``{'version_id': 12345678}``
+        * ``{'account': 'versionista1', 'has_content': True}``
+
+    Returns
+    -------
+    response : dict
+    """
+    params = {'chunk': chunk,
+              'chunk_size': chunk_size,
+              'capture_time': time_range_string(start_date, end_date)}
+    if source_metadata is not None:
+        for k, v in source_metadata.items():
+            params[f'source_metadata[{k}]'] = v
+    url = f'{api_url()}/pages/{page_id}/versions'
+    res = requests.get(url, auth=auth(), params=params)
+    res.raise_for_status()
+    result = res.json()
+    # In place, replace datetime strings with datetime objects.
+    for v in result['data']:
+        v['created_at'] = parse_timestamp(v['created_at'])
+        v['updated_at'] = parse_timestamp(v['updated_at'])
+        v['capture_time'] = parse_timestamp(v['capture_time'])
+    return result
+
+
+@ensure_credentials
+def list_versions(chunk=None, chunk_size=None,
+                  start_date=None, end_date=None,
+                  source_type=None, version_hash=None,
+                  source_metadata=None):
+    """
+    List all Versions, optionally filtered by serach criteria.
+
+    Parameters
+    ----------
+    chunk : integer, optional
+        pagination parameter
+    chunk_size : integer, optional
+        number of items per chunk
+    start_date : datetime, optional
+    end_date : datetime, optional
+    source_type : string, optional
+    version_hash : string, optional
+    source_metadata : dict, optional
+        Examples:
+
+        * ``{'version_id': 12345678}``
+        * ``{'account': 'versionista1', 'has_content': True}``
```

Comment:
Would it make more sense if this and the page-specific version above were the same function with an optional `page=` kwarg? There shouldn’t be any difference between the two as far as the API is concerned.

(In fact, I believe you can GET `/api/v0/versions?page_id={page_id}` and get the same thing as `/api/v0/pages/{page_id}/versions`, though the former should *probably* be discouraged.)